### PR TITLE
Qtbot methods changed to snake case,snake case check added

### PR DESCRIPTION
**Pull request by Lembrun**
Quickfix to #5864
I guess we could do a script to parse the doc to detect new aliases but I feel for now the above should suffice.

### CODE REVIEW

**Review by The-Compiler — CHANGES_REQUESTED** [review]
I think it should be possible to use a regex like `qtbot\.[a-z]+[A-Z][a-zA-Z]*` to check for this (though I haven't tested)?

Also please integrate the check with the existing code (even if it says "spelling") rather than duplicating it.

**Comment by Lembrun** [comment]
I did try a regexp of this kind, and replaced everything but I ran into errors in automated test because some methods do not have snake case aliases (keyPress,MousePress,keyClicks.. see [https://pytest-qt.readthedocs.io/en/latest/_modules/pytestqt/qtbot.html](url),)

I'll merge the rest into spelling while waiting for answer

**Comment by The-Compiler** [comment]
Ah, good point. In that case I'd prefer having a list of those methods which should be ignored, so that when something new gets added to the "hybrid" (snake-case and camel-case API), we don't silently ignore it.

This will need some changes to `misc_checks.py` to have a list of "always allowed" words or so. For now, I'd propose to add a generic regex first (and unify the two functions like mentioned), and we can then take it from there.

**Comment by The-Compiler** [comment]
Actually... I'm okay with both I guess. Keeping it simple certainly has its benefits as well, though I'm happy to assist if you prefer the more generic approach.

If you want to keep the current one, please merge the related methods into one regex using `wait(Active|Exposed|...)`.

**Comment by Lembrun** [comment]
Concerning the generic approach I thought about something like this:

`exclude =  Clicks|Click|Press`

`re.compile(r'qtbot\.[a-z]+(?!{})[A-Z].*'.format(exclude))`

Other regexp  for reference : `qtbot\..*[A-Z].+(?=\()(?<!Clicks)(?<!Press)(?<!Click)` (can't simply do |.. because python says negative look .. non fixed)

What do you think ?

File: scripts/dev/misc_checks.py [hunk 1]
```diff
@@ -175,10 +175,15 @@ def check_spelling(args: argparse.Namespace) -> Optional[bool]:
             "Common misspelling or non-US spelling"
         ) for w in words
     ]
+
+    qtbot_methods = {'keyClick', 'keyClicks', 'mousePress', 'keyPress'}
```
Comment (The-Compiler) [inline scripts/dev/misc_checks.py:179]: I think we should have all methods here, even if we don't use them all.

```suggestion
    qtbot_methods = {
        'keyPress',
        'keyRelease',
        'keyClick',
        'keyClicks',
        'keyEvent',
        'mousePress',
        'mouseRelease',
        'mouseClick',
        'mouseMove',
        'mouseDClick',
        'keySequence'
    }
```

File: scripts/dev/misc_checks.py [hunk 2]
```diff
@@ -175,10 +175,15 @@ def check_spelling(args: argparse.Namespace) -> Optional[bool]:
...
+
+    excluding = '|'.join(qtbot_methods)
```
Comment (The-Compiler) [inline scripts/dev/misc_checks.py:181]: "excluding" is a rather generic name, maybe name it `qtbot_excludes` or so?

File: scripts/dev/misc_checks.py [hunk 3]
```diff
@@ -226,6 +231,10 @@ def check_spelling(args: argparse.Namespace) -> Optional[bool]:
             re.compile(r'IOError'),
             "use OSError",
         ),
+        (
+            re.compile(r'qtbot\.(?!{})[a-z]+[A-Z].*'.format(excluding)),
```
Comment (The-Compiler) [inline scripts/dev/misc_checks.py:235]: You can use [f-strings](https://realpython.com/python-f-strings/) instead:

```suggestion
            re.compile(fr'qtbot\.(?!{excluding})[a-z]+[A-Z].*'),
```

File: scripts/dev/misc_checks.py [hunk 4]
```diff
@@ -226,6 +231,10 @@ def check_spelling(args: argparse.Namespace) -> Optional[bool]:
...
+            "Snake-case available",
```
Comment (The-Compiler) [inline scripts/dev/misc_checks.py:236]:
```suggestion
            "use snake-case instead",
```

to be consistent with the other messages.

**Review by The-Compiler — CHANGES_REQUESTED** [review]

**Comment by The-Compiler** [comment]
Looks great now, thanks! I also changed the script slightly in 2c8dec9ec so it outputs the matched text instead of the (now rather long) pattern.

**Pull request merged**